### DISCUSSION TITLE
Event Listeners and APIs

### DISCUSSION THREAD

**Discussion by dave-english** [scenerystack/community#52] [Q&A] [answered]
I've completed the SceneryStack learning exercises, and now making steady progress on an original project.
I discovered a published [Scenery API here](https://phetsims.github.io/scenery/doc/)
which seems incomplete (the last half dozen links have no information) Event, Pointer, Mouse, Touch, Pen Trail
I am also interested in an API on the Axon library, if anyone can suggest where to find it.

Related to this, there seem to be a lot of different ways to work with listeners.  Some of the PhetSims code I've reviewed call on specific events (drag-listener, etc), whereas others use a ".link" function which appears to call upon a listener buried much deeper in the code. Is there a simple explanation for this, and when to choose one type of construct over another?

And finally for a more specific question, what is the simplest approach to creating a dynamic text output.  Can I take the following code, and embed a listener directly into the options for the new VBox or new HSlider which would cause the second block of text to update when the slider is moved?
```

    const eSlider=new VBox( {
      align: 'center',
      spacing: 10,
      
      children: [
        new Text ( 'Electrode Separation', {font:DEFAULT_FONT,stroke: 'red'}),
        new HSlider( 
          model.separationProperty,
          model.separationProperty.range,
          {constrainValue: value =>Utils.roundSymmetric( value )}
          ),
        new Text ( model.separationProperty.value, { font: DEFAULT_FONT,stroke: 'red', })
          
        ]
     } );
```

**Comment by samreid** [comment, marked as answer]
This shows how to map the separation to a text readout string Property

```ts
    const separationProperty = new NumberProperty( 2 );
    const eSlider = new VBox( {
      align: 'center',
      spacing: 10,

      children: [
        new Text( 'Electrode Separation', { stroke: 'red' } ),
        new HSlider(
          separationProperty,
          new Range( 1, 10 ),
          { constrainValue: value => Utils.roundSymmetric( value ) }
        ),
        new Text( new DerivedProperty( [ separationProperty ], separation => separation.toFixed( 1 ) ), { stroke: 'red' } )
      ]
    } );
    this.addChild( eSlider );
```

We haven't published API docs for axon yet.

Generally, `link` automatically calls back the listener to synchronize things during startup.
